project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module

```diff
@@ -77,6 +99,173 @@ function c4m_message_entity_load($entities, $entity_type) {
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['@group_node_title'] = array(
+    'callback' => 'c4m_message_format_group_node_title',
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['@group_welcome_message'] = array(
+    'callback' => 'c4m_message_format_group_welcome_message',
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['@group_node_type'] = array(
+    'callback' => 'c4m_message_format_group_node_type',
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['!group_dashboard_url'] = array(
+    'callback' => 'c4m_message_format_group_url',
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['!go_public_profile_url'] = array(
+    'callback' => 'c4m_message_format_go_public_profile_url',
+    'pass message' => TRUE,
+  );
+
+  //Add
+  $message->arguments['@go_full_name'] = array(
+    'callback' => 'c4m_message_format_go_full_name',
+    'pass message' => TRUE,
+  );
+}
+
+/**
+ * Return a member's first name.
+ *
+ * @param \Message $message
+ * @return string
+ */
+function c4m_message_format_member_first_name(Message $message) {
+  $account_wrapper = entity_metadata_wrapper('user', $message->uid);
+  $first_name = $account_wrapper->c4m_first_name->value();
+
+  return $first_name ? $first_name : $account_wrapper->label();
+
+}
+
+/**
+ * Return a members's last name.
+ *
+ * @param \Message $message
+ * @return string
+ */
+function c4m_message_format_member_last_name(Message $message) {
+  $account_wrapper = entity_metadata_wrapper('user', $message->uid);
+  $last_name = $account_wrapper->c4m_last_name->value();
+
+  return $last_name ? $last_name : '';
+}
+
+/**
+ * Return group url.
+ *
+ * @param \Message $message
+ * @return string
+ */
+function c4m_message_format_group_url(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $nid = $wrapper->field_group_node->value(array('identifier' => TRUE));
+
+  return url('node/' . $nid);
+}
+
+
+/**
+ * Return group title.
+ *
+ * @param \Message $message
+ * @return string
+ */
+function c4m_message_format_group_node_title(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return $wrapper->field_group_node->label();
+}
+
+/**
+ * Return group welcome message.
+ *
+ * @param \Message $message
+ * @return string
+ *
+ * @todo Add functionality after CFM-1249 is merged
+ */
+function c4m_message_format_group_welcome_message(Message $message) {
+/*  $wrapper = entity_metadata_wrapper('message', $message);
+  $group_message =  $wrapper->field_group_node->field_c4m_welcome_message->value();
+
+  return $group_message ? $group_message : '';*/
+  return '';
+}
+
+/**
+ * Return a group node type.
+ *
+ * @param \Message $message
+ * @return string
+ */
+function c4m_message_format_group_node_type(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $group_type =  $wrapper->field_group_node->type->value();
```

Comment:
@nader77 @amitaibu 
It seems we can simply use `return 'node';` in this callback since _group_ is always a _node_
right ?